diff --git a/extensions/latex-workshop/package.json b/extensions/latex-workshop/package.json
index afe4cbe..90c3dec 100644
--- a/extensions/latex-workshop/package.json
+++ b/extensions/latex-workshop/package.json
@@ -494,6 +494,16 @@
         "title": "%command.hostPort%",
         "category": "LaTeX Workshop"
       },
+      {
+        "command": "underoot.openLaTeXManager",
+        "title": "Open LaTeX Engine Manager",
+        "category": "Underoot"
+      },
+      {
+        "command": "underoot.selectEngine",
+        "title": "Select LaTeX Engine",
+        "category": "Underoot"
+      },
       {
         "command": "latex-workshop.navigate-envpair",
         "title": "%command.navigate-envpair%",
@@ -1972,6 +1982,26 @@
             "Fading transition animation"
           ]
         },
+        "latex-workshop.view.pdf.supersampling": {
+          "scope": "window",
+          "type": "number",
+          "default": 1.0,
+          "minimum": 0.5,
+          "maximum": 4.0,
+          "markdownDescription": "Enable variable supersampling in the PDF viewer. A value of 2.0 provides 2x resolution (crisper text), while 1.0 is standard. Values between 0.5 and 4.0 are supported."
+        },
+        "latex-workshop.view.pdf.gpuRasterization": {
+          "scope": "window",
+          "type": "boolean",
+          "default": false,
+          "markdownDescription": "Enable GPU rasterization for the PDF viewer and editor (requires restart). May improve performance for complex PDFs."
+        },
+        "latex-workshop.view.pdf.lcdTextRendering": {
+          "scope": "window",
+          "type": "boolean",
+          "default": true,
+          "markdownDescription": "Enable LCD text rendering (ClearType) for sharper font display (requires restart)."
+        },
         "latex-workshop.synctex.path": {
           "scope": "window",
           "type": "string",
@@ -3008,4 +3038,4 @@
     "typescript": "^5.7.2",
     "vscode-extend-language": "0.3.0"
   }
-}
+}
\ No newline at end of file
diff --git a/extensions/latex-workshop/viewer/pdfjs/pdf.mjs b/extensions/latex-workshop/viewer/pdfjs/pdf.mjs
index ffdb697..9d158be 100644
--- a/extensions/latex-workshop/viewer/pdfjs/pdf.mjs
+++ b/extensions/latex-workshop/viewer/pdfjs/pdf.mjs
@@ -26,25 +26,30 @@
  */
 /******/ // The require scope
 /******/ var __webpack_require__ = {};
-/******/ 
+/******/
 /************************************************************************/
 /******/ /* webpack/runtime/define property getters */
 /******/ (() => {
 /******/ 	// define getter functions for harmony exports
 /******/ 	__webpack_require__.d = (exports, definition) => {
-/******/ 		for(var key in definition) {
-/******/ 			if(__webpack_require__.o(definition, key) && !__webpack_require__.o(exports, key)) {
+/******/ 		for (var key in definition) {
+/******/ 			if (__webpack_require__.o(definition, key) && !__webpack_require__.o(exports, key)) {
 /******/ 				Object.defineProperty(exports, key, { enumerable: true, get: definition[key] });
-/******/ 			}
-/******/ 		}
-/******/ 	};
-/******/ })();
-/******/ 
+        /******/
+}
+      /******/
+}
+    /******/
+};
+  /******/
+})();
+/******/
 /******/ /* webpack/runtime/hasOwnProperty shorthand */
 /******/ (() => {
 /******/ 	__webpack_require__.o = (obj, prop) => (Object.prototype.hasOwnProperty.call(obj, prop))
-/******/ })();
-/******/ 
+  /******/
+})();
+/******/
 /************************************************************************/
 var __webpack_exports__ = {};
 
@@ -391,7 +396,7 @@ function createValidAbsoluteUrl(url, baseUrl = null, options = null) {
     if (options.tryConvertEncoding) {
       try {
         url = stringToUTF8String(url);
-      } catch {}
+      } catch { }
     }
   }
   const absoluteUrl = baseUrl ? URL.parse(url, baseUrl) : URL.parse(url);
@@ -797,7 +802,7 @@ function stringToPDFString(str, keepEscapeSequence = false) {
   for (let i = 0, ii = str.length; i < ii; i++) {
     const charCode = str.charCodeAt(i);
     if (!keepEscapeSequence && charCode === 0x1b) {
-      while (++i < ii && str.charCodeAt(i) !== 0x1b) {}
+      while (++i < ii && str.charCodeAt(i) !== 0x1b) { }
       continue;
     }
     const code = PDFStringTranslateTable[charCode];
@@ -1660,7 +1665,7 @@ class OutputScale {
     return false;
   }
   static get pixelRatio() {
-    return globalThis.devicePixelRatio || 1;
+    return (globalThis.LW_SUPERSAMPLING || 1) * (globalThis.devicePixelRatio || 1);
   }
   static capPixels(maxPixels, capAreaFactor) {
     if (capAreaFactor >= 0) {
@@ -3789,7 +3794,7 @@ class AnnotationEditorUIManager {
   }
   get #lastSelectedEditor() {
     let ed = null;
-    for (ed of this.#selectedEditors) {}
+    for (ed of this.#selectedEditors) { }
     return ed;
   }
   updateUI(editor) {
@@ -4744,7 +4749,7 @@ class Comment {
     this.#date = null;
     this.#richText = richText;
   }
-  shown() {}
+  shown() { }
   destroy() {
     this.#commentToolbarButton?.remove();
     this.#commentToolbarButton = null;
@@ -5081,7 +5086,7 @@ class AnnotationEditor {
     const style = getComputedStyle(document.documentElement);
     AnnotationEditor._borderLineWidth = parseFloat(style.getPropertyValue("--outline-width")) || 0;
   }
-  static updateDefaultParams(_type, _value) {}
+  static updateDefaultParams(_type, _value) { }
   static get defaultPropertiesToUpdate() {
     return [];
   }
@@ -5261,8 +5266,8 @@ class AnnotationEditor {
       block: "nearest"
     });
   }
-  _onTranslating(x, y) {}
-  _onTranslated(x, y) {}
+  _onTranslating(x, y) { }
+  _onTranslated(x, y) { }
   get _hasBeenMoved() {
     return !!this.#initialRect && (this.#initialRect[0] !== this.x || this.#initialRect[1] !== this.y);
   }
@@ -5483,7 +5488,7 @@ class AnnotationEditor {
     this.fixAndSetPosition();
     this._onResized();
   }
-  _onResized() {}
+  _onResized() { }
   #addResizeToUndoStack() {
     if (!this.#savedDimensions) {
       return;
@@ -5613,7 +5618,7 @@ class AnnotationEditor {
     this.fixAndSetPosition();
     this._onResizing();
   }
-  _onResizing() {}
+  _onResizing() { }
   altTextFinish() {
     this.#altText?.finish();
   }
@@ -5800,7 +5805,7 @@ class AnnotationEditor {
     if (this.hasEditedComment) {
       const DEFAULT_POPUP_WIDTH = 180;
       const DEFAULT_POPUP_HEIGHT = 100;
-      const [,,, trY] = serialized.rect;
+      const [, , , trY] = serialized.rect;
       const [pageWidth] = this.pageDimensions;
       const [pageX] = this.pageTranslation;
       const blX = pageX + pageWidth + 1;
@@ -6008,8 +6013,8 @@ class AnnotationEditor {
       signal
     });
   }
-  _onStartDragging() {}
-  _onStopDragging() {}
+  _onStartDragging() { }
+  _onStopDragging() { }
   moveInDOM() {
     if (this.#moveInDOMTimeout) {
       clearTimeout(this.#moveInDOMTimeout);
@@ -6105,7 +6110,7 @@ class AnnotationEditor {
       opacity
     };
   }
-  onceAdded(focus) {}
+  onceAdded(focus) { }
   isEmpty() {
     return false;
   }
@@ -6163,8 +6168,8 @@ class AnnotationEditor {
   rebuild() {
     this.#addFocusListeners();
   }
-  rotate(_angle) {}
-  resize() {}
+  rotate(_angle) { }
+  resize() { }
   serializeDeleted() {
     return {
       id: this.annotationElementId,
@@ -6430,9 +6435,9 @@ class AnnotationEditor {
       this._uiManager.toggleComment(null);
     }
   }
-  updateParams(type, value) {}
-  disableEditing() {}
-  enableEditing() {}
+  updateParams(type, value) { }
+  disableEditing() { }
+  enableEditing() { }
   get canChangeContent() {
     return false;
   }
@@ -8040,7 +8045,7 @@ const StreamKind = {
   PULL_COMPLETE: 7,
   START_COMPLETE: 8
 };
-function onFn() {}
+function onFn() { }
 function wrapReason(ex) {
   if (ex instanceof AbortException || ex instanceof InvalidPDFException || ex instanceof PasswordException || ex instanceof ResponseException || ex instanceof UnknownErrorException) {
     return ex;
@@ -8547,7 +8552,7 @@ class BaseFilterFactory {
   addHighlightHCMFilter(filterName, fgColor, bgColor, newFgColor, newBgColor) {
     return "none";
   }
-  destroy(keepHCM = false) {}
+  destroy(keepHCM = false) { }
 }
 class DOMFilterFactory extends BaseFilterFactory {
   #baseUrl;
@@ -8947,7 +8952,7 @@ async function node_utils_fetchData(url) {
   const data = await fs.promises.readFile(url);
   return new Uint8Array(data);
 }
-class NodeFilterFactory extends BaseFilterFactory {}
+class NodeFilterFactory extends BaseFilterFactory { }
 class NodeCanvasFactory extends BaseCanvasFactory {
   _createCanvas(width, height) {
     const require = process.getBuiltinModule("module").createRequire(import.meta.url);
@@ -10852,8 +10857,8 @@ class CanvasGraphics {
       ctx.lineDashOffset = dashPhase;
     }
   }
-  setRenderingIntent(opIdx, intent) {}
-  setFlatness(opIdx, flatness) {}
+  setRenderingIntent(opIdx, intent) { }
+  setFlatness(opIdx, flatness) { }
   setGState(opIdx, states) {
     for (const [key, value] of states) {
       switch (key) {
@@ -11611,7 +11616,7 @@ class CanvasGraphics {
       this.dependencyTracker = dependencyTracker;
     }
   }
-  setCharWidth(opIdx, xWidth, yWidth) {}
+  setCharWidth(opIdx, xWidth, yWidth) { }
   setCharWidthAndBounds(opIdx, xWidth, yWidth, llx, lly, urx, ury) {
     const clip = new Path2D();
     clip.rect(llx, lly, urx - llx, ury - lly);
@@ -12094,8 +12099,8 @@ class CanvasGraphics {
     this.ctx.fillRect(0, 0, 1, 1);
     this.compose();
   }
-  markPoint(opIdx, tag) {}
-  markPointProps(opIdx, tag, properties) {}
+  markPoint(opIdx, tag) { }
+  markPointProps(opIdx, tag, properties) { }
   beginMarkedContent(opIdx, tag) {
     this.dependencyTracker?.beginMarkedContent(opIdx);
     this.markedContentStack.push({
@@ -12120,8 +12125,8 @@ class CanvasGraphics {
     this.markedContentStack.pop();
     this.contentVisible = this.isContentVisible();
   }
-  beginCompat(opIdx) {}
-  endCompat(opIdx) {}
+  beginCompat(opIdx) { }
+  endCompat(opIdx) { }
   consumePath(opIdx, path, clipBox) {
     const isEmpty = this.current.isEmptyClip();
     if (this.pendingClip) {
@@ -12874,7 +12879,7 @@ function getFilenameFromContentDispositionHeader(contentDisposition) {
         const buffer = stringToBytes(value);
         value = decoder.decode(buffer);
         needsEncodingFixup = false;
-      } catch {}
+      } catch { }
     }
     return value;
   }
@@ -12958,7 +12963,7 @@ function getFilenameFromContentDispositionHeader(contentDisposition) {
       }
       try {
         text = atob(text);
-      } catch {}
+      } catch { }
       return textdecode(charset, text);
     });
   }
@@ -13023,7 +13028,7 @@ function extractFilenameFromHeader(responseHeaders) {
     if (filename.includes("%")) {
       try {
         filename = decodeURIComponent(filename);
-      } catch {}
+      } catch { }
     }
     if (isPdfFile(filename)) {
       return filename;
@@ -14001,7 +14006,7 @@ class TextLayer {
       TextLayer.#pendingTextLayers.delete(this);
       this.#layoutTextParams = null;
       this.#styleCache = null;
-    }).catch(() => {});
+    }).catch(() => { });
   }
   static get fontFamilyMap() {
     const {
@@ -14061,7 +14066,7 @@ class TextLayer {
   }
   cancel() {
     const abortEx = new AbortException("TextLayer task cancelled.");
-    this.#reader?.cancel(abortEx).catch(() => {});
+    this.#reader?.cancel(abortEx).catch(() => { });
     this.#reader = null;
     this.#capability.reject(abortEx);
   }
@@ -14576,7 +14581,7 @@ class PDFDataRangeTransport {
   requestDataRange(begin, end) {
     unreachable("Abstract method PDFDataRangeTransport.requestDataRange");
   }
-  abort() {}
+  abort() { }
 }
 class PDFDocumentProxy {
   constructor(pdfInfo, transport) {
@@ -15124,7 +15129,7 @@ class PDFPageProxy {
         return;
       }
     }
-    intentState.streamReader.cancel(new AbortException(reason.message)).catch(() => {});
+    intentState.streamReader.cancel(new AbortException(reason.message)).catch(() => { });
     intentState.streamReader = null;
     if (this._transport.destroyed) {
       return;
@@ -15212,7 +15217,7 @@ class PDFWorker {
   #initializeFromPort(port) {
     this.#port = port;
     this.#messageHandler = new MessageHandler("main", "worker", port);
-    this.#messageHandler.on("ready", () => {});
+    this.#messageHandler.on("ready", () => { });
     this.#resolve();
   }
   #initialize() {
@@ -15342,9 +15347,9 @@ class PDFWorker {
         return this.#mainThreadWorkerMessageHandler;
       }
       const worker = await import(
-      /*webpackIgnore: true*/
-      /*@vite-ignore*/
-      this.workerSrc);
+        /*webpackIgnore: true*/
+        /*@vite-ignore*/
+        this.workerSrc);
       return worker.WorkerMessageHandler;
     };
     return shadow(this, "_setupFakeWorkerGlobal", loader());
@@ -15960,7 +15965,7 @@ class InternalRenderTask {
     this._operationsFilter = operationsFilter;
   }
   get completed() {
-    return this.capability.promise.catch(function () {});
+    return this.capability.promise.catch(function () { });
   }
   initializeGraphics({
     transparency = false,
@@ -16346,7 +16351,7 @@ class BasicColorPicker {
     this.#input?.remove();
     this.#input = null;
   }
-  hideDropdown() {}
+  hideDropdown() { }
 }
 
 ;// ./src/shared/scripting_utils.js
@@ -18349,7 +18354,7 @@ class ChoiceWidgetAnnotationElement extends WidgetAnnotationElement {
     }
     if (this.data.combo) {
       this._setTextStyle(selectElement);
-    } else {}
+    } else { }
     this._setBackgroundColor(selectElement);
     this._setDefaultPropertiesFromJS(selectElement);
     this.container.append(selectElement);
@@ -21388,7 +21393,7 @@ class HighlightEditor extends AnnotationEditor {
         break;
     }
   }
-  translateInPage(x, y) {}
+  translateInPage(x, y) { }
   get toolbarPosition() {
     return this.#lastPoint;
   }
@@ -22442,7 +22447,7 @@ class DrawingEditor extends AnnotationEditor {
     }
     this.#updateBbox(this.#drawOutlines.updateParentDimensions(this.parentDimensions, this.parent.scale));
   }
-  static onScaleChangingWhenDrawing() {}
+  static onScaleChangingWhenDrawing() { }
   render() {
     if (this.div) {
       return this.div;
@@ -22643,7 +22648,7 @@ class DrawingEditor extends AnnotationEditor {
     this._cleanup(true);
     return null;
   }
-  createDrawingOptions(_data) {}
+  createDrawingOptions(_data) { }
   static deserializeDraw(_pageX, _pageY, _pageWidth, _pageHeight, _innerWidth, _data) {
     unreachable("Not implemented");
   }
@@ -23179,7 +23184,7 @@ class InkDrawOutline extends Outline {
     };
   }
   get rotationTransform() {
-    const [,, width, height] = this.#bbox;
+    const [, , width, height] = this.#bbox;
     let a = 0,
       b = 0,
       c = 0,
@@ -24149,7 +24154,7 @@ class SignatureExtractor {
       writer.write(bytes).then(async () => {
         await writer.ready;
         await writer.close();
-      }).catch(() => {});
+      }).catch(() => { });
       let data = null;
       let offset = 0;
       for await (const chunk of readable) {
@@ -24673,7 +24678,7 @@ class StampEditor extends AnnotationEditor {
       });
       try {
         this.mlGuessAltText();
-      } catch {}
+      } catch { }
     }
     this.div.focus();
   }
